From c6fb4b27032c61709fedbb18820486aa17292026 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 20 Jul 2026 14:42:17 +0200
Subject: [PATCH] Fix delete-event callback return type

The main window's delete-event handler was declared with a void return
type even though GTK expects a gboolean return value.

Return TRUE to prevent GTK from destroying the window when hiding the
window to the system tray.
---
 src/gui.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gui.c b/src/gui.c
index 892ac8c..4644cdf 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -224,7 +224,7 @@ gui_quit_osmo (GUI *appGUI) {
 
 /*------------------------------------------------------------------------------*/
 
-void
+gboolean
 gui_window_close_cb (GtkWidget *widget, GdkEvent *event, gpointer user_data) {
 
     GUI *appGUI = (GUI *)user_data;
@@ -234,6 +234,8 @@ gui_window_close_cb (GtkWidget *widget, GdkEvent *event, gpointer user_data) {
     } else {
         gui_quit_osmo (appGUI);
     }
+
+    return TRUE;
 }
 
 /*------------------------------------------------------------------------------*/
-- 
2.55.0

